### DISCUSSION TITLE
RTL feature

### DISCUSSION THREAD

**Discussion by YoussefZidan** [tailwindlabs/tailwindcss#5716] [Ideas]
Like Bootstrap5 adding RTL feature and substitute left, right classes with start, end classes.

**Comment by simonswiss** [comment]
Hey!

This is something we have considered a few times, but honestly don't have enough expertise/experience building RTL websites to be able to integrate something like this with confidence.

There is a pretty good community plugin available here, you can play with the live demo to see what it does!

https://www.npmjs.com/package/tailwindcss-rtl

![68747470733a2f2f692e696d6775722e636f6d2f515359386e4e552e676966](https://user-images.githubusercontent.com/485747/136301268-1f7c2596-f7f0-497e-8797-f58ef84ac02f.gif)

**Comment by adamwathan** [comment]
We actually have experimental RTL support in the Tailwind v3.0 alpha! We aren't using logical properties yet and instead using `rtl` and `ltr` modifiers, but hoping to explore logical property support in the future. Biggest challenge is just naming things honestly.

Here's an example of what you can do in Tailwind 3:

```html
<div class="pl-4 rtl:pr-4">
  ...
</div>
```

Need to add this to the release notes, kinda forgot about it 🤦‍♂️

**Reply by dcastil** [reply]
What does speak against adding classes like [`inline-padding-start`](https://developer.mozilla.org/en-US/docs/Web/CSS/padding-inline-start) (for `padding-left`) etc. from the tailwind-rtl plugin? With JIT it shouldn't create any overhead, right?

**Reply by vltansky** [reply]
@adamwathan maybe even use CSS logical properties looks like it's already widely supported: https://caniuse.com/css-logical-props?

https://github.com/stevecochrane/tailwindcss-logical

**Comment by Reex11** [comment]
I've used this plugin, its great!
It uses classes like: `start-x` and `end-x` without ditching `left-x` and `right-x`

I very much like this approach because it will automatically decide the direction without having to specify both RTL and LTR. Also, Devs will be able to force left or right if I want by using the default `left-x` and `right-x`.
An Example: 

Original Code:
```
<div class="text-green border pl-4 mr-1 right-0 rounded-l-md">
```
RTL support Code:
```
<div class="text-green border ps-4 me-1 end-0 rounded-s-md">
```
This way we can support RTL, and the original code would work as expected without any change (if you're don't care about RTL).
Also, I can force left padding in both RTL and LTR layouts if I want like this:
```
<div class="text-green border pl-4 me-1 end-0 rounded-s-md">
```

This is more elegant, easier to read and to implement.

**Comment by alaadahmed** [comment]
Hi everyone, currently I am working on big project and depend extensively on `rtl` and `ltr` utilities. I hope they are stable enough. So far they are working great except very few times I need to refresh the page to make these utilities take effect. Web app is __English__ <-> __Arabic__ 

https://user-images.githubusercontent.com/14099508/150695248-219f6164-0676-450a-8ac9-13a10135836f.mov
